dpnp/dpnp_iface_arraycreation.py

```diff
@@ -612,15 +612,16 @@ def full(shape, fill_value, dtype=None, order='C'):
 
     """
 
-    if (not use_origin_backend()):
-        if order not in ('C', 'c', None):
-            checker_throw_value_error("full", "order", order, 'C')
-
-        _dtype = dtype if dtype is not None else type(fill_value)
+    if not use_origin_backend():
+        if order != 'C':
```

Comment:
```suggestion
        if order is not 'C':
```

Reply:
Generally strings in Python are compared by value, not identity. I have one example which shows a potential issue:
```python
a = 'hi'
b = ''.join(['h', 'i'])
a == b  # True
a is b  # False
```